WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubTimeRangeController.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+import Yosemite
+
+struct TimeRange {
+    let start: Date
+    let end: Date
+}
+
+public class AnalyticsHubTimeRangeController {
```

Comment:
I think it is worth adding some documentation about this type. Lets aim to have our types as documented as possible! 💪 

Additionally, I see that the main purpose of this type is to generate ranges, so what about calling it `AnalyticsHubRangeGenerator` or  `AnalyticsHubRangeGenerator`. We normally reserve the word `Controller` for the `UIKit` View Controller 😁

Reply:
Good point @Ecarrion! Let me fix that! Just one question: maybe we could call that type a `AnalyticsHubTimeRangeSet`? The type is a data set of two time ranges and their respective descriptions. If we want to use it as a generator, maybe we should avoid making the type retain a state and leave just functions to create and return the full set. WDYT?

Reply:
I used the word `Generator` because all of the functions begin with the prefix `generate...`. 

I'm not sure I understand the `Set` idea, to me, `Set` refers to the `Set` data structure 😅

Reply:
I see. I mentioned `set` because the interface of the type could be consumed as a set of two TimeRanges. But I see that this can lead to confusion. Let's adopt the generator word then. Thanks!

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8193/commits/61188082b5b99a1f147fc4ac9e686c114855ef32